x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/EnterpriseSearch.java

```diff
@@ -119,6 +122,16 @@ public Collection<Object> createComponents(
             // TODO We need to use use a real BigArrays which recycles pages here
             BigArrays.NON_RECYCLING_INSTANCE
         );
+
+        // Behavioral analytics components
+        final AnalyticsTemplateRegistry analyticsTemplateRegistry = new AnalyticsTemplateRegistry(
+            clusterService,
+            threadPool,
+            client,
+            xContentRegistry
+        );
+        analyticsTemplateRegistry.initialize();
+
         return Collections.singletonList(engineService);
```

Comment:
Should we add the registry in the returned components?

Reply:
I do not think it exposes method that are meant to be used out of the scope of initialization so I did not returned it. I can change it or merge the current version. Let me know what your preference is.

Reply:
Let's add it to be in sync with the other modules, thanks.

Reply:
I change it and the analytics registry is now returned by the method.